### ISSUE TITLE
[BUILD] RPCS3 fails to build the LLVM submodule when host uses lates LLVM 12.0.0: error: Record `Commutative' does not have a field named `IsDefault'!

### ISSUE THREAD

**Issue by SupervisedThinking** [RPCS3/rpcs3#10117]
## Quick summary
RPCS3 https://github.com/RPCS3/rpcs3/commit/99d6f9c7a91acec0905fdd72e8fe19aadcbc4960 fails to build the LLVM submodule when the host uses LLVM 12.

## Details

```
Executing (target): ninja 
[0/2] Re-checking globbed directories...
[40/1965] Building IntrinsicImpl.inc...
FAILED: 3rdparty/llvm_build/include/llvm/IR/IntrinsicImpl.inc 
cd /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/.x86_64-libreelec-linux-gnu && /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/toolchain/bin/llvm-tblgen -gen-intrinsic-impl -I /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/llvm/include/llvm/IR -I /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/libxml2 -I /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/.x86_64-libreelec-linux-gnu/3rdparty/llvm_build/include -I /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/llvm/include /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/llvm/include/llvm/IR/Intrinsics.td --write-if-changed -o 3rdparty/llvm_build/include/llvm/IR/IntrinsicImpl.inc -d 3rdparty/llvm_build/include/llvm/IR/IntrinsicImpl.inc.d
/mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/llvm/include/llvm/IR/Intrinsics.td:60:1: error: Record `Commutative' does not have a field named `IsDefault'!

def Commutative : IntrinsicProperty;
^

```
Please describe the regression as accurately as possible.

Build log: https://pastebin.com/Y4act55S

#### 6. Please provide your system configuration:
- LibreELEC 10.x fork https://github.com/SupervisedThinking/LibreELEC-RR/tree/master-rr
- RPCS3 package https://github.com/SupervisedThinking/LibreELEC-RR/blob/master-rr/packages/supervisedthinking/emulation/standalone/rpcs3/package.mk

**Comment by hcorion** [comment]
Try removing this line from your package.mk
```
                         -D LLVM_TABLEGEN=${TOOLCHAIN}/bin/llvm-tblgen \
```

**Comment by SupervisedThinking** [comment]
> Try removing this line from your package.mk and do a clean build
> 
> ```
>                          -D LLVM_TABLEGEN=${TOOLCHAIN}/bin/llvm-tblgen \
> ```

You're aware that I'm compiling an embedded build that builds it's own toolchain & LLVM? It compiles fine with LLVM 11.1.0 so the RPCS3 LLVM fork needs probably updates or fixes.

**Comment by xddxd** [comment]
Why are you using your toolchain’s tblgen? The submodule builds it’s own and uses that.

**Comment by hcorion** [comment]
So the thing here is that our fork will never and has never attempted to be compatible with parts of the system LLVM.

**Comment by SupervisedThinking** [comment]
> Why are you using your toolchain’s tblgen? The submodule builds it’s own and uses that.

Nope it won't. It's using the toolchain tblgen to build the target llvm submodule for the package and since target uses newer version `GLIBC_2.32' not found you can't run that at build time & I have to use llvm-tblgen build for the host/toolchain

```
-- Build files have been written to: /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/.x86_64-libreelec-linux-gnu/3rdparty/llvm_build/NATIVE
[792/2040] Building native llvm-tblgen...
[211/211] Linking CXX executable bin/llvm-tblgen
[796/2040] Building IntrinsicImpl.inc...
FAILED: 3rdparty/llvm_build/include/llvm/IR/IntrinsicImpl.inc 
cd /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/.x86_64-libreelec-linux-gnu && /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/.x86_64-libreelec-linux-gnu/3rdparty/llvm_build/NATIVE/bin/llvm-tblgen -gen-intrinsic-impl -I /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/llvm/include/llvm/IR -I /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/libxml2 -I /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/.x86_64-libreelec-linux-gnu/3rdparty/llvm_build/include -I /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/llvm/include /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/llvm/include/llvm/IR/Intrinsics.td --write-if-changed -o 3rdparty/llvm_build/include/llvm/IR/IntrinsicImpl.inc -d 3rdparty/llvm_build/include/llvm/IR/IntrinsicImpl.inc.d
/mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/.x86_64-libreelec-linux-gnu/3rdparty/llvm_build/NATIVE/bin/llvm-tblgen: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/.x86_64-libreelec-linux-gnu/3rdparty/llvm_build/NATIVE/bin/llvm-tblgen)
[797/2040] Building Attributes.inc...
FAILED: 3rdparty/llvm_build/include/llvm/IR/Attributes.inc 
cd /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/.x86_64-libreelec-linux-gnu && /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/.x86_64-libreelec-linux-gnu/3rdparty/llvm_build/NATIVE/bin/llvm-tblgen -gen-attrs -I /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/llvm/include/llvm/IR -I /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/libxml2 -I /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/.x86_64-libreelec-linux-gnu/3rdparty/llvm_build/include -I /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/llvm/include /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/llvm/include/llvm/IR/Attributes.td --write-if-changed -o 3rdparty/llvm_build/include/llvm/IR/Attributes.inc -d 3rdparty/llvm_build/include/llvm/IR/Attributes.inc.d
/mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/.x86_64-libreelec-linux-gnu/3rdparty/llvm_build/NATIVE/bin/llvm-tblgen: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/.x86_64-libreelec-linux-gnu/3rdparty/llvm_build/NATIVE/bin/llvm-tblgen)
[798/2040] Building IntrinsicsAArch64.h...
FAILED: 3rdparty/llvm_build/include/llvm/IR/IntrinsicsAArch64.h 
cd /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/.x86_64-libreelec-linux-gnu && /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/.x86_64-libreelec-linux-gnu/3rdparty/llvm_build/NATIVE/bin/llvm-tblgen -gen-intrinsic-enums -intrinsic-prefix=aarch64 -I /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/llvm/include/llvm/IR -I /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/libxml2 -I /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/.x86_64-libreelec-linux-gnu/3rdparty/llvm_build/include -I /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/llvm/include /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/llvm/include/llvm/IR/Intrinsics.td --write-if-changed -o 3rdparty/llvm_build/include/llvm/IR/IntrinsicsAArch64.h -d 3rdparty/llvm_build/include/llvm/IR/IntrinsicsAArch64.h.d
/mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/.x86_64-libreelec-linux-gnu/3rdparty/llvm_build/NATIVE/bin/llvm-tblgen: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/.x86_64-libreelec-linux-gnu/3rdparty/llvm_build/NATIVE/bin/llvm-tblgen)
[799/2040] Building IntrinsicsBPF.h...
FAILED: 3rdparty/llvm_build/include/llvm/IR/IntrinsicsBPF.h 
cd /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/.x86_64-libreelec-linux-gnu && /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/.x86_64-libreelec-linux-gnu/3rdparty/llvm_build/NATIVE/bin/llvm-tblgen -gen-intrinsic-enums -intrinsic-prefix=bpf -I /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/llvm/include/llvm/IR -I /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/libxml2 -I /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/.x86_64-libreelec-linux-gnu/3rdparty/llvm_build/include -I /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/llvm/include /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/llvm/include/llvm/IR/Intrinsics.td --write-if-changed -o 3rdparty/llvm_build/include/llvm/IR/IntrinsicsBPF.h -d 3rdparty/llvm_build/include/llvm/IR/IntrinsicsBPF.h.d
/mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/.x86_64-libreelec-linux-gnu/3rdparty/llvm_build/NATIVE/bin/llvm-tblgen: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /mnt/dev/LibreELEC-RR/build.LibreELEC-Generic.x86_64-10.0-devel/build/rpcs3-99d6f9c7a91acec0905fdd72e8fe19aadcbc4960/.x86_64-libreelec-linux-gnu/3rdparty/llvm_build/NATIVE/bin/llvm-tblgen)
ninja: build stopped: subcommand failed.
FAILURE: scripts/build rpcs3 during make_target (default)
*********** FAILED COMMAND ***********
ninja ${NINJA_OPTS} ${PKG_MAKE_OPTS_TARGET}
**************************************
```

Also the submodule is missing a test directory & fails to compile when it reaches https://github.com/RPCS3/llvm-mirror/blob/master/CMakeLists.txt#L1061

**Comment by SupervisedThinking** [comment]
> So the thing here is that our fork will never and has never attempted to be compatible with parts of the system LLVM.

I'm pretty sure this problem is related to the fact that you still use LLVM 11.x - I'm building RPCS3 for years now this way.

**Comment by xddxd** [comment]
I honestly have no idea what you did here. I was able to compile rpcs3 and llvm in Ubuntu 18.04 which uses glibc 2.27, with gcc 10 and clang 12.

**Comment by SupervisedThinking** [comment]
compile =! cross compile. 

If you follow the links you would understand that I'm talking about building a system image for an embedded system. So I use a host system to build a complete (!) toolchain which later builds the target packages. Your glibc 2.27 belongs to the Ubuntu host system which isn't used later on the target system.

Anyway I did not know that:

> So the thing here is that our fork will never and has never attempted to be compatible with parts of the system LLVM.

It worked flawless until now to reuse the llvm-tblgen of the toolchain but obviously LLVM 11 & 12 don't play along.

Anyway I've solved it by splitting the RPCS3 build process into a host & target system, I build the llvm-tblgen based on your fork first with host flags, so e.g. it would be linked against glibc 2.27, and point the package to the fork llvm-tblgen instead of toolchain llvm-tblgen used to build LLVM 12 for the target system.
